### PR TITLE
External Contribution: Use RenderTexture instead of Camera for Visual Observation

### PR DESCRIPTION
**Pull request by ervteng**
@pyjamads
#1698

### CODE REVIEW

File: docs/Learning-Environment-Design-Agents.md [hunk 1]
```diff
@@ -200,29 +200,53 @@ used in your normalization formula.
...
 In addition, make sure that the Agent's Brain expects a visual observation. In
 the Brain inspector, under **Brain Parameters** > **Visual Observations**,
-specify the number of Cameras the Agent is using for its visual observations.
+specify the number of Resolutions the Agent is using for its visual observations.
 For each visual observation, set the width and height of the image (in pixels)
 and whether or not the observation is color or grayscale (when `Black And White`
-is checked).
+is checked). If a combination of `Cameras` and `RenderTextures` is used, all 
+cameras are captured first, then all render textures will be added, both in the
+order they appear in the editor. With 2 cameras and 1 render texture of say twice
```
Comment (vincentpierre) [inline docs/Learning-Environment-Design-Agents.md]: I think for numbers like this, we need to say "one" and "two" not 1 and 2. 
Also, Why is the render texture using a different resolution ? This is confusing.

File: UnitySDK/Assets/ML-Agents/Examples/GridWorld/Scenes/GridWorldRenderTexture.unity [hunk 2]
```diff
@@ -0,0 +1,1301 @@
+%YAML 1.1
```
Comment (vincentpierre) [inline UnitySDK/Assets/ML-Agents/Examples/GridWorld/Scenes/GridWorldRenderTexture.unity]: Control Checkbox needs to be unchecked.

File: docs/Learning-Environment-Design-Agents.md [hunk 3]
```diff
@@ -200,29 +200,53 @@ used in your normalization formula.
...
+
+![Agent Camera and RenderTexture combination](images/visual-observation-combination.png)
+
+RenderTexture observations will throw an `Exception` if the width/height, doesn't 
+match the resolution specified under **Brain Parameters** > **Visual Observations**.
+
+When using `RenderTexture` visual observations a handy feature for debugging, is 
+adding a `Canvas` with a `RawImage` with it's texture set to the `RenderTexture`, 
+in order to view the agent observation. The [GridWorld example](Learning-Environment-Examples.md#gridworld) 
+GridWorldRenderTexture scene uses a RenderTexure for debugging and observation.
```
Comment (vincentpierre) [inline docs/Learning-Environment-Design-Agents.md]: There is no explanation on how to update the renderTexture. I think there needs to be a note and maybe a screenshot on how to drag the renderTexture into the field of the camera. <-- This is not obvious.

File: UnitySDK/Assets/ML-Agents/Examples/GridWorld/Scenes/GridWorldRenderTexture.unity [hunk 4]
```diff
@@ -0,0 +1,1301 @@
...
+%TAG !u! tag:unity3d.com,2011:
```
Comment (vincentpierre) [inline UnitySDK/Assets/ML-Agents/Examples/GridWorld/Scenes/GridWorldRenderTexture.unity]: Not a fan of having 2 gridworld scenes now... This adds confusion and testing requirements.

Comment (ervteng) [reply UnitySDK/Assets/ML-Agents/Examples/GridWorld/Scenes/GridWorldRenderTexture.unity]: Would it be preferable to have a new example, or no example at all?

File: docs/Learning-Environment-Design-Agents.md (hunk 3 above)
Comment (ervteng) [reply docs/Learning-Environment-Design-Agents.md]: Added screenshot and reworded this paragraph

File: UnitySDK/Assets/ML-Agents/Examples/GridWorld/Scenes/GridWorldRenderTexture.unity (hunk 2 above)
Comment (ervteng) [reply UnitySDK/Assets/ML-Agents/Examples/GridWorld/Scenes/GridWorldRenderTexture.unity]: Unchecked!

File: docs/Learning-Environment-Design-Agents.md (hunk 1 above)
Comment (ervteng) [reply docs/Learning-Environment-Design-Agents.md]: Reworded this whole section to make it clearer.

**Pull request merged**